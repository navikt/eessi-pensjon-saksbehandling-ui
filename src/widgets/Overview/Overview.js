import React, { useState, useEffect } from 'react'
import PT from 'prop-types'
import _ from 'lodash'
import { connect, bindActionCreators } from 'store'
import * as appActions from 'actions/app'
import { Nav } from 'eessi-pensjon-ui'
import { getDisplayName } from 'utils/displayName'
import PersonTitle from './PersonTitle'
import PersonPanel from './PersonPanel'

import './Overview.css'

const mapStateToProps = (state) => {
  return {
    aktoerId: state.app.params.aktoerId,
    gettingPersonInfo: state.loading.gettingPersonInfo,
    isSendingPinfo: state.loading.isSendingPinfo,
    locale: state.ui.locale,
    person: state.app.person,
    sakId: state.app.params.sakId
  }
}

const mapDispatchToProps = (dispatch) => {
  return { actions: bindActionCreators({ ...appActions }, dispatch) }
}

export const Overview = (props) => {
  const { actions, aktoerId, onUpdate, widget } = props
  const [mounted, setMounted] = useState(false)

  useEffect(() => {
    if (!mounted && aktoerId) {
      actions.getPersonInfo(aktoerId)
      setMounted(true)
    }
  }, [mounted, actions, aktoerId])

  const onExpandablePanelChange = () => {
    const newWidget = _.cloneDeep(widget)
    newWidget.options.collapsed = !newWidget.options.collapsed
    onUpdate(newWidget)
  }

  return (
    <Nav.EkspanderbartpanelBase
      className='w-overview s-border'
      apen={!widget.options.collapsed}
      onClick={onExpandablePanelChange}
      heading={<PersonTitle {...props} />}
    >
      <PersonPanel {...props} />
    </Nav.EkspanderbartpanelBase>
  )
}

Overview.propTypes = {
  actions: PT.object.isRequired,
  aktoerId: PT.string,
  onUpdate: PT.func.isRequired,
  widget: PT.object.isRequired
}

const ConnectedOverview = connect(mapStateToProps, mapDispatchToProps)(Overview)
ConnectedOverview.displayName = `Connect(${getDisplayName(Overview)})`
export default ConnectedOverview
