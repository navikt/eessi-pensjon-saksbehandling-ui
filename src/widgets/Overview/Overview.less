.w-overview-personPanel {
  &__waiting,
  &__title {
    display: flex;
    align-items: center;
  }
  &__content {
    columns: 3;

    .grey {
      color: grey;
    }
  }
}
