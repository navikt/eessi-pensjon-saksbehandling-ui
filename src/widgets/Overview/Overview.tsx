import { getPersonAvdodInfo, getPersonInfo } from 'actions/app'
import classNames from 'classnames'
import { HighContrastExpandingPanel } from 'components/StyledComponents'
import * as constants from 'constants/constants'
import { WidgetPropType } from 'declarations/Dashboard.pt'
import { State } from 'declarations/reducers'
import { AllowedLocaleString, FeatureToggles, PesysContext } from 'declarations/app.d'
import { PersonAvdods } from 'declarations/person.d'
import _ from 'lodash'
import { standardLogger, timeDiffLogger } from 'metrics/loggers'
import { Widget } from 'nav-dashboard'
import Alertstripe from 'nav-frontend-alertstriper'
import { theme, themeHighContrast } from 'nav-styled-component-theme'
import PT from 'prop-types'
import React, { useEffect, useState } from 'react'
import { useTranslation } from 'react-i18next'
import { useDispatch, useSelector } from 'react-redux'
import styled, { ThemeProvider } from 'styled-components'
import PersonPanel from './PersonPanel'
import PersonTitle from './PersonTitle'

export const Alert = styled(Alertstripe)`
  width: 100%
`

export interface OverviewSelector {
  aktoerId: string
  featureToggles: FeatureToggles
  gettingPersonInfo: boolean
  locale: AllowedLocaleString
  person: any,
  personAvdods: PersonAvdods | undefined,
  pesysContext: PesysContext | undefined,
  vedtakId: string
}

const mapState = (state: State): OverviewSelector => ({
  /* istanbul ignore next */
  aktoerId: state.app.params.aktoerId,
  featureToggles: state.app.featureToggles,
  gettingPersonInfo: state.loading.gettingPersonInfo,
  locale: state.ui.locale,
  person: state.app.person,
  personAvdods: state.app.personAvdods,
  pesysContext: state.app.pesysContext,
  vedtakId: state.app.params.vedtakId
})

export interface OverviewProps {
  highContrast: boolean
  onUpdate?: (w: Widget) => void
  skipMount?: boolean
  widget: Widget
}

export const Overview: React.FC<OverviewProps> = ({
  highContrast,
  onUpdate,
  skipMount = false,
  widget
}: OverviewProps): JSX.Element => {
  const [mounted, setMounted] = useState<boolean>(skipMount)
  const { aktoerId, featureToggles, gettingPersonInfo, locale, person, personAvdods, pesysContext, vedtakId }: OverviewSelector =
    useSelector<State, OverviewSelector>(mapState)
  const [totalTimeWithMouseOver, setTotalTimeWithMouseOver] = useState<number>(0)
  const [mouseEnterDate, setMouseEnterDate] = useState<Date | undefined>(undefined)

  useEffect(() => {
    return () => {
      timeDiffLogger('overview.mouseover', totalTimeWithMouseOver)
    }
  }, [])

  const dispatch = useDispatch()
  const { t } = useTranslation()

  const onExpandablePanelClosing = (): void => {
    const newWidget = _.cloneDeep(widget)
    newWidget.options.collapsed = true
    standardLogger('overview.ekspandpanel.close')
    if (onUpdate) {
      onUpdate(newWidget)
    }
  }

  const onExpandablePanelOpening = (): void => {
    const newWidget = _.cloneDeep(widget)
    newWidget.options.collapsed = false
    standardLogger('overview.ekspandpanel.open')
    if (onUpdate) {
      onUpdate(newWidget)
    }
  }

  const onMouseEnter = () => setMouseEnterDate(new Date())

  const onMouseLeave = () => {
    if (mouseEnterDate) {
      setTotalTimeWithMouseOver(totalTimeWithMouseOver + (new Date().getTime() - mouseEnterDate?.getTime()))
    }
  }

  useEffect(() => {
    if (!mounted && aktoerId && pesysContext) {
      if (!person) {
        dispatch(getPersonInfo(aktoerId))
        if (pesysContext === constants.VEDTAKSKONTEKST) {
          dispatch(getPersonAvdodInfo(aktoerId, vedtakId, featureToggles.NR_AVDOD))
        }
      }
      setMounted(true)
    }
  }, [featureToggles, mounted, dispatch, aktoerId, person, pesysContext, vedtakId])

  if (!aktoerId) {
    return (
      <Alert
        type='advarsel'
        data-test-id='w-overview__alert'
      >
        {t('buc:validation-noAktoerId')}
      </Alert>
    )
  }

  return (
    <ThemeProvider theme={highContrast ? themeHighContrast : theme}>
      <div
        onMouseEnter={onMouseEnter}
        onMouseLeave={onMouseLeave}
      >
        <HighContrastExpandingPanel
          highContrast={highContrast}
          collapseProps={{ id: 'w-overview-id' }}
          className={classNames({ highContrast: highContrast })}
          data-test-id='w-overview-id'
          open={!widget.options.collapsed}
          onOpen={onExpandablePanelOpening}
          onClose={onExpandablePanelClosing}
          heading={(
            <PersonTitle
              gettingPersonInfo={gettingPersonInfo}
              person={person}
            />
          )}
        >
          <PersonPanel
            highContrast={highContrast}
            locale={locale}
            person={person}
            personAvdods={personAvdods}
          />
        </HighContrastExpandingPanel>
      </div>
    </ThemeProvider>
  )
}

Overview.propTypes = {
  onUpdate: PT.func,
  skipMount: PT.bool,
  widget: WidgetPropType.isRequired
}

export default Overview
