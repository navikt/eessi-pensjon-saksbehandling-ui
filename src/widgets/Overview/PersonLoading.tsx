import LoadingImage from 'components/Loading/LoadingImage'
import LoadingText from 'components/Loading/LoadingText'
import { HorizontalSeparatorDiv } from 'components/StyledComponents'
import React from 'react'
import styled from 'styled-components'

const Title = styled.div`
  display: flex;
  align-items: center;
  flex: 1;
`
const PlaceholderText = styled(LoadingText)`
   height: 40px;
   min-height: 40px;
   width: 100%;
`
const PersonLoading = () => {
  return (
    <Title>
      <LoadingImage />
      <HorizontalSeparatorDiv />
      <PlaceholderText />
    </Title>
  )
}

export default PersonLoading
