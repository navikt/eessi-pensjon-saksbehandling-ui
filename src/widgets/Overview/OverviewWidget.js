import React, { useState, useEffect } from 'react'
import PT from 'prop-types'
import ReactResizeDetector from 'react-resize-detector'
import Overview from './Overview'

const OverviewWidget = ({ onResize, onUpdate, widget }) => {
  const [mounted, setMounted] = useState(false)

  useEffect(() => {
    if (!mounted && onResize) {
      onResize()
      setMounted(true)
    }
  }, [mounted, onResize])

  return (
    <div className='w-OverviewWidget'>
      <ReactResizeDetector
        handleWidth
        handleHeight
        onResize={onResize}
      />
      <Overview onUpdate={onUpdate} widget={widget} />
    </div>
  )
}

OverviewWidget.properties = {
  type: 'overview',
  title: 'Overview widget',
  description: 'Widget with overview info',
  layout: {
    lg: { minW: 6, maxW: 12, defaultW: 6, minH: 2, defaultH: 4, maxH: 999 },
    md: { minW: 3, maxW: 3, defaultW: 1, minH: 2, defaultH: 4, maxH: 999 },
    sm: { minW: 1, maxW: 1, defaultW: 1, minH: 2, defaultH: 4, maxH: 999 }
  },
  options: {
    collapsed: false
  }
}

OverviewWidget.propTypes = {
  onResize: PT.func.isRequired,
  onUpdate: PT.func.isRequired,
  widget: PT.object.isRequired
}

export default OverviewWidget
