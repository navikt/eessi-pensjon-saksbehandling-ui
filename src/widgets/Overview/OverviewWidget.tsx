import { WidgetPropType } from 'declarations/Dashboard.pt'
import { WidgetFC, WidgetProps } from 'eessi-pensjon-ui/dist/declarations/Dashboard'
import PT from 'prop-types'
import React, { useEffect, useState } from 'react'
import ReactResizeDetector from 'react-resize-detector'
import Overview from './Overview'

const OverviewWidget: WidgetFC<WidgetProps> = ({
  onResize, onUpdate, widget
}: WidgetProps): JSX.Element => {
  const [mounted, setMounted] = useState<boolean>(false)

  useEffect(() => {
    if (!mounted && onResize) {
      onResize()
      setMounted(true)
    }
  }, [mounted, onResize])

  return (
    <div className='w-OverviewWidget'>
      <ReactResizeDetector
        handleWidth
        handleHeight
        onResize={onResize}
      />
      <Overview onUpdate={onUpdate} widget={widget} />
    </div>
  )
}

OverviewWidget.properties = {
  type: 'overview',
  title: 'Overview widget',
  description: 'Widget with overview info',
  layout: {
    lg: { minW: 6, maxW: 12, defaultW: 6, minH: 2, defaultH: 4, maxH: 999 },
    md: { minW: 3, maxW: 3, defaultW: 1, minH: 2, defaultH: 4, maxH: 999 },
    sm: { minW: 1, maxW: 1, defaultW: 1, minH: 2, defaultH: 4, maxH: 999 }
  },
  options: {
    collapsed: false
  }
}

OverviewWidget.propTypes = {
  onResize: PT.func.isRequired,
  onUpdate: PT.func.isRequired,
  widget: WidgetPropType.isRequired
}

export default OverviewWidget
