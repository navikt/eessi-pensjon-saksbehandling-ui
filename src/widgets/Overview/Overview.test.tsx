import { mount, ReactWrapper } from 'enzyme'
import React from 'react'
import { Overview, OverviewProps } from './Overview'

describe('widgets/Overview/Overview', () => {
  let wrapper: ReactWrapper
  const initialMockProps: OverviewProps = {
    actions: {
      getPersonInfo: jest.fn()
    },
    aktoerId: '123',
    highContrast: false,
    locale: 'nb',
    onUpdate: jest.fn(),
    t: jest.fn(t => t),
    widget: {
      options: {
        collapsed: false
      }
    }
  }

  beforeEach(() => {
    wrapper = mount(<Overview {...initialMockProps} />)
  })

  afterEach(() => {
    wrapper.unmount()
  })

  it('Renders', () => {
    expect(wrapper.isEmptyRender()).toBeFalsy()
  })

  it('UseEffect: fetches person info when mounting', () => {
    expect(initialMockProps.actions.getPersonInfo).toHaveBeenCalled()
  })

  it('Has proper HTML structure', () => {
    expect(wrapper.exists('.w-overview')).toBeTruthy()
    expect(wrapper.exists('EkspanderbartpanelBase')).toBeTruthy()
    expect(wrapper.exists('PersonTitle')).toBeTruthy()
    expect(wrapper.exists('PersonPanel')).toBeTruthy()
  })

  it('With no aktoerId', () => {
    wrapper = mount(<Overview {...initialMockProps} aktoerId={undefined} />)
    expect(wrapper.exists('.w-overview__alert')).toBeTruthy()
    expect(wrapper.find('.w-overview__alert').hostNodes().render().text()).toEqual('buc:validation-noAktoerId')
  })

  it('Expandable ', () => {
    wrapper.find('EkspanderbartpanelBase button').simulate('click')
    expect(initialMockProps.onUpdate).toHaveBeenCalledWith({
      options: {
        collapsed: true
      }
    })
  })
})
