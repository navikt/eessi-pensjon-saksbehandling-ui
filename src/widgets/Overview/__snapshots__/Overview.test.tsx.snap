// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`widgets/Overview/Overview Renders 1`] = `
.c3.ekspanderbartPanel--apen .ReactCollapse--collapse {
  will-change: max-height,height;
  max-height: 150em;
  -webkit-animation: iZaYLF 250ms ease;
  animation: iZaYLF 250ms ease;
}

.c3.ekspanderbartPanel--lukket .ReactCollapse--collapse {
  will-change: max-height,height;
  max-height: 0;
  -webkit-animation: hIThqZ 250ms ease;
  animation: hIThqZ 250ms ease;
}

ui-fss.g1[id="c2"] {
  content: "c3,";
}

.c9 {
  display: inline-block;
  margin-left: 1rem;
}

ui-fss.g14[id="c8"] {
  content: "c9,";
}

.c1 {
  border-radius: 4px;
  background-color: #FFFFFF;
  border-width: 1px;
  border-style: solid;
  border-color: #78706A;
}

.c1 .ekspanderbartPanel__knapp {
  background: none;
  border: none;
}

.c1.ekspanderbartPanel {
  background-color: #FFFFFF;
}

.c1 .ekspanderbartPanel__hode {
  background-color: #FFFFFF;
}

.c1 .ekspanderbartPanel__hode:hover {
  background-color: #e0f4fb !important;
}

.c1 .ekspanderbartPanel__indikator {
  color: #3E3832;
}

.c1 p {
  font-size: inherit;
  line-height: inherit;
}

ui-fss.g24[id="c0"] {
  content: "c1,";
}

.c7 {
  min-width: 40px;
  min-height: 40px;
  background-color: whitesmoke;
  border-radius: 100px;
  -webkit-transition: width .6s ease;
  transition: width .6s ease;
  -webkit-animation: da-daAO 1s linear infinite;
  animation: da-daAO 1s linear infinite;
  background-image: linear-gradient(45deg,hsla(90,90%,90%,.15) 25%,transparent 0,transparent 50%,hsla(90,90%,90%,.15) 0,hsla(90,90%,90%,.15) 75%,transparent 0,transparent);
}

ui-fss.g31[id="c6"] {
  content: "c7,";
}

.c13 {
  min-width: 50%;
  -webkit-transition: width .6s ease;
  transition: width .6s ease;
  background-color: whitesmoke;
  -webkit-animation: da-daAO 1s linear infinite;
  animation: da-daAO 1s linear infinite;
  background-image: linear-gradient(45deg,hsla(90,90%,90%,.15) 25%,transparent 0,transparent 50%,hsla(90,90%,90%,.15) 0,hsla(90,90%,90%,.15) 75%,transparent 0,transparent);
}

ui-fss.g32[id="c12"] {
  content: "c13,";
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
}

ui-fss.g33[id="c4"] {
  content: "c5,";
}

.c11 {
  height: 40px;
  min-height: 40px;
  width: 100%;
}

ui-fss.g34[id="c10"] {
  content: "c11,";
}

<Overview
  onUpdate={[MockFunction]}
  skipMount={false}
  widget={
    Object {
      "i": "i",
      "options": Object {
        "collapsed": false,
      },
      "title": "foo",
      "type": "foo",
      "visible": true,
    }
  }
>
  <Component
    theme={
      Object {
        "alternative-background-color": "#E9E7E7",
        "black": "#000000",
        "border-radius-base": "4px",
        "border-radius-large": "6px",
        "border-radius-small": "2px",
        "container-desktop": "(940px + @grid-gutter-width)",
        "container-large-desktop": "(1140px + @grid-gutter-width)",
        "container-lg": "(1140px + @grid-gutter-width)",
        "container-md": "(940px + @grid-gutter-width)",
        "container-sm": "(720px + @grid-gutter-width)",
        "container-tablet": "(720px + @grid-gutter-width)",
        "fokusFarge": "#254a6c",
        "grayBackground": "#E9E7E7",
        "grayIcon": "#B7B1A9",
        "grayInactive": "#78706A",
        "grayModia": "#333333",
        "grid-columns": "12",
        "grid-float-breakpoint": "768px",
        "grid-float-breakpoint-max": "(@grid-float-breakpoint - 1)",
        "grid-gutter-width": "16px",
        "inverted-font-color": "#FFFFFF",
        "line-height-large": "1.3333333",
        "line-height-small": "1.5",
        "main-active-color": "#a3dff3",
        "main-background-color": "#FFFFFF",
        "main-border-color": "#78706A",
        "main-border-radius": "4px",
        "main-disabled-color": "#B7B1A9",
        "main-error-color": "#BA3A26",
        "main-focus-color": "#254a6c",
        "main-font-color": "#3E3832",
        "main-hover-color": "#e0f4fb",
        "main-interactive-color": "#0067C5",
        "main-success-color": "#06893A",
        "main-warning-color": "#FF9100",
        "navBla": "#0067C5",
        "navBlaDarken20": "#0c5da7",
        "navBlaDarken40": "#18548a",
        "navBlaDarken60": "#254a6c",
        "navBlaDarken80": "#31414f",
        "navBlaLighten20": "#3385d0",
        "navBlaLighten40": "#66a3dc",
        "navBlaLighten60": "#99c2e7",
        "navBlaLighten80": "#cce0f3",
        "navDypBla": "#005B82",
        "navDypBlaDarken20": "#0c5472",
        "navDypBlaDarken40": "#184d62",
        "navDypBlaDarken60": "#254652",
        "navDypBlaDarken80": "#313f42",
        "navDypBlaLighten20": "#337b9b",
        "navDypBlaLighten40": "#669cb4",
        "navDypBlaLighten60": "#99bdcd",
        "navDypBlaLighten80": "#ccdee6",
        "navGra20": "#C6C2BF",
        "navGra40": "#B7B1A9",
        "navGra60": "#78706A",
        "navGra80": "#59514B",
        "navGronn": "#06893A",
        "navGronnDarken20": "#117838",
        "navGronnDarken40": "#1c6836",
        "navGronnDarken60": "#275835",
        "navGronnDarken80": "#324833",
        "navGronnLighten20": "#37a061",
        "navGronnLighten40": "#69b888",
        "navGronnLighten60": "#9bcfb0",
        "navGronnLighten80": "#cde7d7",
        "navLilla": "#634689",
        "navLillaDarken20": "#5b4377",
        "navLillaDarken40": "#544066",
        "navLillaDarken60": "#4c3d54",
        "navLillaDarken80": "#453a43",
        "navLillaLighten20": "#826ba0",
        "navLillaLighten40": "#a190b8",
        "navLillaLighten60": "#c0b5cf",
        "navLillaLighten80": "#dfdae7",
        "navLimeGronn": "#A2AD00",
        "navLimeGronnDarken20": "#8e950a",
        "navLimeGronnDarken40": "#7a7e14",
        "navLimeGronnDarken60": "#66661e",
        "navLimeGronnDarken80": "#524f28",
        "navLimeGronnLighten20": "#b4bd33",
        "navLimeGronnLighten40": "#c7cd66",
        "navLimeGronnLighten60": "#d9de99",
        "navLimeGronnLighten80": "#eceecc",
        "navLysBla": "#66CBEC",
        "navLysBlaDarken20": "#5eadc6",
        "navLysBlaDarken40": "#5690a1",
        "navLysBlaDarken60": "#4e727c",
        "navLysBlaDarken80": "#465557",
        "navLysBlaLighten20": "#84d5ef",
        "navLysBlaLighten40": "#a3dff3",
        "navLysBlaLighten60": "#c1eaf7",
        "navLysBlaLighten80": "#e0f4fb",
        "navLysGra": "#E9E7E7",
        "navMorkGra": "#3E3832",
        "navOransje": "#FF9100",
        "navOransjeDarken20": "#d87f0a",
        "navOransjeDarken40": "#b16d14",
        "navOransjeDarken60": "#8b5b1e",
        "navOransjeDarken80": "#644928",
        "navOransjeLighten20": "#ffa733",
        "navOransjeLighten40": "#ffbd66",
        "navOransjeLighten60": "#ffd399",
        "navOransjeLighten80": "#ffe9cc",
        "navRod": "#C30000",
        "navRodDarken20": "#a80b0a",
        "navRodDarken40": "#8d1614",
        "navRodDarken60": "#73211e",
        "navRodDarken80": "#582c28",
        "navRodLighten20": "#cf3333",
        "navRodLighten40": "#db6666",
        "navRodLighten60": "#e79999",
        "navRodLighten80": "#f3cccc",
        "orangeFocus": "#FFBD66",
        "orangeFocusDarken20": "#d8a25b",
        "orangeFocusDarken40": "#b18751",
        "orangeFocusDarken60": "#8b6d46",
        "orangeFocusDarken80": "#64523c",
        "orangeFocusLighten20": "#ffca84",
        "orangeFocusLighten40": "#ffd7a3",
        "orangeFocusLighten60": "#ffe4c1",
        "orangeFocusLighten80": "#fff1e0",
        "padding-base-horizontal": "12px",
        "padding-base-vertical": "6px",
        "padding-large-horizontal": "16px",
        "padding-large-vertical": "10px",
        "padding-small-horizontal": "10px",
        "padding-small-vertical": "5px",
        "padding-xs-horizontal": "5px",
        "padding-xs-vertical": "1px",
        "pinkErrorBg": "#F3E3E3",
        "redError": "#BA3A26",
        "redErrorDarken20": "#a13928",
        "redErrorDarken40": "#88392a",
        "redErrorDarken60": "#6f382d",
        "redErrorDarken80": "#56382f",
        "redErrorLighten20": "#c76151",
        "redErrorLighten40": "#d5887c",
        "redErrorLighten60": "#e3b0a8",
        "redErrorLighten80": "#f1d7d3",
        "screen-desktop": "992px",
        "screen-lg": "1200px",
        "screen-lg-desktop": "1200px",
        "screen-lg-min": "1200px",
        "screen-md": "992px",
        "screen-md-max": "(@screen-lg-min - 1)",
        "screen-md-min": "992px",
        "screen-sm": "768px",
        "screen-sm-max": "(@screen-md-min - 1)",
        "screen-sm-min": "768px",
        "screen-tablet": "768px",
        "screen-xs-max": "(@screen-sm-min - 1)",
        "type": "theme",
        "white": "#FFFFFF",
      }
    }
  >
    <div
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
      <Styled(ExpandingPanel)
        className=""
        collapseProps={
          Object {
            "id": "w-overview-id",
          }
        }
        heading={
          <PersonTitle
            gettingPersonInfo={false}
          />
        }
        onClick={[Function]}
        open={true}
      >
        <ExpandingPanel
          className="c0 c1"
          collapseProps={
            Object {
              "id": "w-overview-id",
            }
          }
          heading={
            <PersonTitle
              gettingPersonInfo={false}
            />
          }
          onClick={[Function]}
          open={true}
        >
          <Component
            theme={
              Object {
                "alternative-background-color": "#E9E7E7",
                "black": "#000000",
                "border-radius-base": "4px",
                "border-radius-large": "6px",
                "border-radius-small": "2px",
                "container-desktop": "(940px + @grid-gutter-width)",
                "container-large-desktop": "(1140px + @grid-gutter-width)",
                "container-lg": "(1140px + @grid-gutter-width)",
                "container-md": "(940px + @grid-gutter-width)",
                "container-sm": "(720px + @grid-gutter-width)",
                "container-tablet": "(720px + @grid-gutter-width)",
                "fokusFarge": "#254a6c",
                "grayBackground": "#E9E7E7",
                "grayIcon": "#B7B1A9",
                "grayInactive": "#78706A",
                "grayModia": "#333333",
                "grid-columns": "12",
                "grid-float-breakpoint": "768px",
                "grid-float-breakpoint-max": "(@grid-float-breakpoint - 1)",
                "grid-gutter-width": "16px",
                "inverted-font-color": "#FFFFFF",
                "line-height-large": "1.3333333",
                "line-height-small": "1.5",
                "main-active-color": "#a3dff3",
                "main-background-color": "#FFFFFF",
                "main-border-color": "#78706A",
                "main-border-radius": "4px",
                "main-disabled-color": "#B7B1A9",
                "main-error-color": "#BA3A26",
                "main-focus-color": "#254a6c",
                "main-font-color": "#3E3832",
                "main-hover-color": "#e0f4fb",
                "main-interactive-color": "#0067C5",
                "main-success-color": "#06893A",
                "main-warning-color": "#FF9100",
                "navBla": "#0067C5",
                "navBlaDarken20": "#0c5da7",
                "navBlaDarken40": "#18548a",
                "navBlaDarken60": "#254a6c",
                "navBlaDarken80": "#31414f",
                "navBlaLighten20": "#3385d0",
                "navBlaLighten40": "#66a3dc",
                "navBlaLighten60": "#99c2e7",
                "navBlaLighten80": "#cce0f3",
                "navDypBla": "#005B82",
                "navDypBlaDarken20": "#0c5472",
                "navDypBlaDarken40": "#184d62",
                "navDypBlaDarken60": "#254652",
                "navDypBlaDarken80": "#313f42",
                "navDypBlaLighten20": "#337b9b",
                "navDypBlaLighten40": "#669cb4",
                "navDypBlaLighten60": "#99bdcd",
                "navDypBlaLighten80": "#ccdee6",
                "navGra20": "#C6C2BF",
                "navGra40": "#B7B1A9",
                "navGra60": "#78706A",
                "navGra80": "#59514B",
                "navGronn": "#06893A",
                "navGronnDarken20": "#117838",
                "navGronnDarken40": "#1c6836",
                "navGronnDarken60": "#275835",
                "navGronnDarken80": "#324833",
                "navGronnLighten20": "#37a061",
                "navGronnLighten40": "#69b888",
                "navGronnLighten60": "#9bcfb0",
                "navGronnLighten80": "#cde7d7",
                "navLilla": "#634689",
                "navLillaDarken20": "#5b4377",
                "navLillaDarken40": "#544066",
                "navLillaDarken60": "#4c3d54",
                "navLillaDarken80": "#453a43",
                "navLillaLighten20": "#826ba0",
                "navLillaLighten40": "#a190b8",
                "navLillaLighten60": "#c0b5cf",
                "navLillaLighten80": "#dfdae7",
                "navLimeGronn": "#A2AD00",
                "navLimeGronnDarken20": "#8e950a",
                "navLimeGronnDarken40": "#7a7e14",
                "navLimeGronnDarken60": "#66661e",
                "navLimeGronnDarken80": "#524f28",
                "navLimeGronnLighten20": "#b4bd33",
                "navLimeGronnLighten40": "#c7cd66",
                "navLimeGronnLighten60": "#d9de99",
                "navLimeGronnLighten80": "#eceecc",
                "navLysBla": "#66CBEC",
                "navLysBlaDarken20": "#5eadc6",
                "navLysBlaDarken40": "#5690a1",
                "navLysBlaDarken60": "#4e727c",
                "navLysBlaDarken80": "#465557",
                "navLysBlaLighten20": "#84d5ef",
                "navLysBlaLighten40": "#a3dff3",
                "navLysBlaLighten60": "#c1eaf7",
                "navLysBlaLighten80": "#e0f4fb",
                "navLysGra": "#E9E7E7",
                "navMorkGra": "#3E3832",
                "navOransje": "#FF9100",
                "navOransjeDarken20": "#d87f0a",
                "navOransjeDarken40": "#b16d14",
                "navOransjeDarken60": "#8b5b1e",
                "navOransjeDarken80": "#644928",
                "navOransjeLighten20": "#ffa733",
                "navOransjeLighten40": "#ffbd66",
                "navOransjeLighten60": "#ffd399",
                "navOransjeLighten80": "#ffe9cc",
                "navRod": "#C30000",
                "navRodDarken20": "#a80b0a",
                "navRodDarken40": "#8d1614",
                "navRodDarken60": "#73211e",
                "navRodDarken80": "#582c28",
                "navRodLighten20": "#cf3333",
                "navRodLighten40": "#db6666",
                "navRodLighten60": "#e79999",
                "navRodLighten80": "#f3cccc",
                "orangeFocus": "#FFBD66",
                "orangeFocusDarken20": "#d8a25b",
                "orangeFocusDarken40": "#b18751",
                "orangeFocusDarken60": "#8b6d46",
                "orangeFocusDarken80": "#64523c",
                "orangeFocusLighten20": "#ffca84",
                "orangeFocusLighten40": "#ffd7a3",
                "orangeFocusLighten60": "#ffe4c1",
                "orangeFocusLighten80": "#fff1e0",
                "padding-base-horizontal": "12px",
                "padding-base-vertical": "6px",
                "padding-large-horizontal": "16px",
                "padding-large-vertical": "10px",
                "padding-small-horizontal": "10px",
                "padding-small-vertical": "5px",
                "padding-xs-horizontal": "5px",
                "padding-xs-vertical": "1px",
                "pinkErrorBg": "#F3E3E3",
                "redError": "#BA3A26",
                "redErrorDarken20": "#a13928",
                "redErrorDarken40": "#88392a",
                "redErrorDarken60": "#6f382d",
                "redErrorDarken80": "#56382f",
                "redErrorLighten20": "#c76151",
                "redErrorLighten40": "#d5887c",
                "redErrorLighten60": "#e3b0a8",
                "redErrorLighten80": "#f1d7d3",
                "screen-desktop": "992px",
                "screen-lg": "1200px",
                "screen-lg-desktop": "1200px",
                "screen-lg-min": "1200px",
                "screen-md": "992px",
                "screen-md-max": "(@screen-lg-min - 1)",
                "screen-md-min": "992px",
                "screen-sm": "768px",
                "screen-sm-max": "(@screen-md-min - 1)",
                "screen-sm-min": "768px",
                "screen-tablet": "768px",
                "screen-xs-max": "(@screen-sm-min - 1)",
                "type": "theme",
                "white": "#FFFFFF",
              }
            }
          >
            <styled.div
              className="ekspanderbartPanel c0 c1 ekspanderbartPanel--apen"
              style={Object {}}
            >
              <div
                className="c2 c3 ekspanderbartPanel c0 c1 ekspanderbartPanel--apen"
                style={Object {}}
              >
                <div
                  aria-controls="w-overview-id"
                  aria-expanded={true}
                  className="ekspanderbartPanel__hode"
                  onClick={[Function]}
                >
                  <div
                    className="ekspanderbartPanel__flex-wrapper"
                  >
                    <PersonTitle
                      gettingPersonInfo={false}
                    >
                      <PersonLoading>
                        <styled.div>
                          <div
                            className="c4 c5"
                          >
                            <_default>
                              <styled.div>
                                <div
                                  className="c6 c7"
                                />
                              </styled.div>
                            </_default>
                            <styled.div>
                              <div
                                className="c8 c9"
                              />
                            </styled.div>
                            <Styled(_default)>
                              <_default
                                className="c10 c11"
                              >
                                <styled.div
                                  className="c10 c11"
                                >
                                  <div
                                    className="c12 c13 c10 c11"
                                  />
                                </styled.div>
                              </_default>
                            </Styled(_default)>
                          </div>
                        </styled.div>
                      </PersonLoading>
                    </PersonTitle>
                    <button
                      aria-expanded={true}
                      aria-label="open"
                      className="ekspanderbartPanel__knapp"
                      onClick={[Function]}
                      onKeyDown={[Function]}
                      type="button"
                    >
                      <span
                        className="ekspanderbartPanel__indikator"
                      />
                    </button>
                  </div>
                </div>
                <UnmountClosed
                  id="w-overview-id"
                  isOpened={true}
                  onRest={[Function]}
                >
                  <Collapse
                    checkTimeout={50}
                    id="w-overview-id"
                    initialStyle={
                      Object {
                        "height": "auto",
                        "overflow": "initial",
                      }
                    }
                    isOpened={true}
                    onRest={[Function]}
                    onWork={[Function]}
                    theme={
                      Object {
                        "collapse": "ReactCollapse--collapse",
                        "content": "ReactCollapse--content",
                      }
                    }
                  >
                    <div
                      className="ReactCollapse--collapse"
                      style={
                        Object {
                          "height": "auto",
                          "overflow": "initial",
                        }
                      }
                    >
                      <div
                        className="ReactCollapse--content"
                      >
                        <article
                          className="ekspanderbartPanel__innhold"
                        >
                          <PersonPanel
                            locale="nb"
                          />
                        </article>
                      </div>
                    </div>
                  </Collapse>
                </UnmountClosed>
              </div>
            </styled.div>
          </Component>
        </ExpandingPanel>
      </Styled(ExpandingPanel)>
    </div>
  </Component>
</Overview>
`;
